app/models/concerns/eds_subjects.rb

```diff
@@ -0,0 +1,99 @@
+module EdsSubjects
+  def eds_subjects
+    @eds_subjects ||= Subject.from(self['eds_subjects'])
+  end
+
+  def eds_subjects_geographic
+    @eds_subjects_geographic ||= Subject.from(self['eds_subjects_geographic'])
+  end
+
+  def eds_subjects_person
+    @eds_subjects_person ||= Subject.from(self['eds_subjects_person'])
+  end
+
+  def eds_author_supplied_keywords
+    @eds_author_supplied_keywords ||= Subject.from(self['eds_author_supplied_keywords'])
+  end
+
+  # EDS-centric subject with linking capability
+  class Subject
+    attr_reader :terms
+    delegate :present?, to: :terms
+
+    class Term < Struct.new(:fieldCode, :searchTerm, :label)
+      delegate :to_s, to: :label
+
+      def to_html
+        field = case fieldCode
+                when 'SH'
+                  'subject_heading'
+                when 'DE'
+                  'descriptor'
+                when 'SU'
+                  'subject'
+                when 'KW'
+                  'keyword'
+                else
+                  'subject'
+                end
+        # TODO: we don't have a view context for `link_to`
+        path = Rails.application.routes.url_helpers.articles_path
+        "<a href=\"#{path}?q=#{CGI.escape(searchTerm)}&amp;search_field=#{field}\">#{CGI.escapeHTML(label)}</a>"
+      end
+    end
+
+    def initialize(terms)
+      @terms = terms
+    end
+
+    def to_s
+      terms.map(&:to_s).join(' -- ')
+    end
+
+    def to_html
+      terms.map(&:to_html).join(' -- ')
+    end
+
+    # @return [Array<Subject>]
+    def self.from(string_or_array)
+      return [] if string_or_array.blank?
+      xml = case string_or_array
+            when String
+              convert_simple_string_to_xml(string_or_array)
+            when Array
+              string_or_array.map { |s| convert_simple_string_to_xml(s) }.join('<br/>')
```

Comment:
This might be a little more simply written as

```
Array.wrap(string_or_array).map { |s| convert_simple_string_to_xml(s) }.join('<br/>')
```

That would allow us to drop the `case` entirely I believe.